### PR TITLE
Expand dz_field tests, add more coords support

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR:
 - Expands `dz_field` tests
 - Adds `dz_field` support for more coords types

I've been running into situations in ClimaAtmos when unifying edmf and dycore radiation pieces where `z_component(::Type{<:Geometry.XZPoint}) = 4` is needed. I figured it would be better to just add support across the board.

It'd be good to confirm that I'm interpreting `coordinate_axis` correctly (which was needed for some of the tests).

### CODE REVIEW

**Comment by valeriabarra** [comment]
Thanks Charlie for adding/expanding these. I've seen this uses the local geometry info. Does it mean that this uses the generalized coordinates only and we don't need to test this on both uniform and stretched meshes?

**Comment by charleskawczynski** [comment]
> Thanks Charlie for adding/expanding these. I've seen this uses the local geometry info. Does it mean that this uses the generalized coordinates only and we don't need to test this on both uniform and stretched meshes?

Yes, this should work for uniform or stretched meshes AFAIU.

File: src/Spaces/finitedifference.jl [hunk 1]
```diff
@@ -161,7 +161,12 @@ The index of the z-component of an abstract point
 in an `AxisTensor`.
 """
 z_component(::Type{<:Geometry.LatLongZPoint}) = 9
```
Comment (simonbyrne) [inline src/Spaces/finitedifference.jl:163]: should this be deltaz component?

Comment (charleskawczynski) [reply src/Spaces/finitedifference.jl:163]: I'm fine with renaming it if you thing `dz_component` is a better name. It's used in `getproperty(lg.∂x∂ξ.components.data, z_component(data_layout_type))`, so I interpreted that as the `z`-component of `dx` (the multi-dimensional vector)

Comment (simonbyrne) [reply src/Spaces/finitedifference.jl:163]: `Δz_metric_component`

File: src/Spaces/finitedifference.jl [hunk 2]
```diff
@@ -161,7 +161,12 @@ The index of the z-component of an abstract point
...
+z_component(::Type{<:Geometry.Cartesian3Point}) = 1
+z_component(::Type{<:Geometry.Cartesian13Point}) = 4
+z_component(::Type{<:Geometry.Cartesian123Point}) = 9
+z_component(::Type{<:Geometry.XYZPoint}) = 9
 z_component(::Type{<:Geometry.ZPoint}) = 1
+z_component(::Type{<:Geometry.XZPoint}) = 4
 
 """
     dz_data(space::AbstractSpace)
```
Comment (simonbyrne) [inline src/Spaces/finitedifference.jl:172]:
```suggestion
    Δz_data(space::AbstractSpace)
```

Comment (simonbyrne) [reply src/Spaces/finitedifference.jl:172]: and add a deprecation

**Comment by charleskawczynski** [comment]
Ran into 
```
--------------------------------------------------------------------------
--
  | An ORTE daemon has unexpectedly failed after launch and before
  | communicating back to mpirun. This could be caused by a number
  | of factors, including an inability to create a connection back
  | to mpirun due to a lack of common network interfaces and/or no
  | route found between them. Please check network connectivity
  | (including firewalls and network routing requirements).
  | --------------------------------------------------------------------------
```

**Comment by charleskawczynski** [comment]
Grr, now ran into:

> 🚨 Error: The command was interrupted by a signal

https://buildkite.com/clima/climacore-ci/builds/1542#0183f74f-5646-4e36-9b0e-b99f04ebcc34/333-475

**Pull request merged**